File: BMIApp/app/src/main/java/pl/pjwstk/bmiapp/utils/NavigationHelper.java
package pl.pjwstk.bmiapp.utils;

import androidx.navigation.NavController;
import androidx.navigation.NavOptions;

import lombok.experimental.UtilityClass;
import pl.pjwstk.bmiapp.R;

@UtilityClass
public final class NavigationHelper {

    public boolean handleNavigation(int itemId, NavController navController) {
        NavOptions navOptions = new NavOptions.Builder()
                .setLaunchSingleTop(true)
                .setPopUpTo(navController.getGraph().getStartDestination(), false)
                .build();

        if (itemId == R.id.homeFragment) {
            navController.navigate(R.id.homeFragment, null, navOptions);
            return true;
        } else if (itemId == R.id.bmiCalculatorFragment) {
            navController.navigate(R.id.bmiCalculatorFragment, null, navOptions);
            return true;
        } else if (itemId == R.id.calorieCalculatorFragment) {
            navController.navigate(R.id.calorieCalculatorFragment, null, navOptions);
            return true;
        }
        return false;
    }
}

File: BMIApp/app/src/main/java/pl/pjwstk/bmiapp/utils/NavigationHelper.kt
package pl.pjwstk.bmiapp.utils

import android.os.Bundle
import android.util.Log
import androidx.navigation.NavController
import androidx.navigation.NavOptions
import pl.pjwstk.bmiapp.R

object NavigationHelper {
    private const val TAG = "NavigationHelper"

    fun handleNavigation(itemId: Int, navController: NavController): Boolean {
        val navOptions = NavOptions.Builder()
            .setLaunchSingleTop(true)
            .setPopUpTo(navController.graph.startDestinationId, false)
            .build()

        when (itemId) {
            R.id.homeFragment -> {
                navController.navigate(R.id.homeFragment, null, navOptions)
                return true
            }

            R.id.bmiCalculatorFragment -> {
                navController.navigate(R.id.bmiCalculatorFragment, null, navOptions)
                return true
            }

            R.id.calorieCalculatorFragment -> {
                navController.navigate(R.id.calorieCalculatorFragment, null, navOptions)
                return true
            }

            R.id.shoppingListFragment -> {
                navController.navigate(R.id.shoppingListFragment, null, navOptions)
                return true
            }

            else -> return false
        }
    }

    fun navigateToShoppingList(navController: NavController, recipeId: Int) {
        Log.d(TAG, "Nawigacja do listy zakupów z recipeId: $recipeId")

        val bundle = Bundle()
        bundle.putInt("recipeId", recipeId)

        navController.navigate(R.id.shoppingListFragment, bundle)
    }
}